spec/lib/efile/nj/nj_property_tax_eligibility_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe Efile::Nj::NjPropertyTaxEligibility do
+  describe ".determine_eligibility" do
+    [
+      { filing_status: :single, income: 10_001, expected: :not_ineligible },
+      { filing_status: :married_filing_jointly, income: 20_001, expected: :not_ineligible },
+      { filing_status: :married_filing_separately, income: 10_001, expected: :not_ineligible },
+      { filing_status: :head_of_household, income: 20_001, expected: :not_ineligible },
+      { filing_status: :qualifying_widow, income: 20_001, expected: :not_ineligible },
+
+      { filing_status: :single, income: 10_000, primary_over_65: true, expected: :ineligible_for_deduction },
+      { filing_status: :married_filing_jointly, income: 20_000, primary_blind: true, expected: :ineligible_for_deduction },
+      { filing_status: :married_filing_separately, income: 10_000, primary_disabled: true, expected: :ineligible_for_deduction },
+      { filing_status: :head_of_household, income: 20_000, primary_over_65: true, expected: :ineligible_for_deduction },
+      { filing_status: :qualifying_widow, income: 20_000, primary_blind: true, expected: :ineligible_for_deduction },
+
+      { filing_status: :married_filing_jointly, income: 19_999, spouse_over_65: true, expected: :ineligible_for_deduction },
+      { filing_status: :married_filing_jointly, income: 19_999, spouse_blind: true, expected: :ineligible_for_deduction },
+      { filing_status: :married_filing_jointly, income: 19_999, spouse_disabled: true, expected: :ineligible_for_deduction },
+
+      { filing_status: :single, income: 10_000, expected: :ineligible },
+      { filing_status: :married_filing_jointly, income: 20_000, expected: :ineligible },
+      { filing_status: :married_filing_separately, income: 10_000, expected: :ineligible },
+      { filing_status: :head_of_household, income: 20_000, expected: :ineligible },
+      { filing_status: :qualifying_widow, income: 20_000, expected: :ineligible },
+
+      { filing_status: :single, income: 9_999, expected: :ineligible },
+      { filing_status: :married_filing_jointly, income: 19_999, expected: :ineligible },
+      { filing_status: :married_filing_separately, income: 9_999, expected: :ineligible },
+      { filing_status: :head_of_household, income: 19_999, expected: :ineligible },
+      { filing_status: :qualifying_widow, income: 19_999, expected: :ineligible },
+    ].each do |test_case|
+      context "when filing #{test_case}" do
+        let(:intake) do
+          traits = []
+          traits << test_case[:filing_status] if test_case[:filing_status] != :single
```

Comment:
[pebble] Making the shape of the objects above something like { traits: { traitx: value }, expected: blah } would allow you to ditch lines 36-44

Reply:
good idea-- will do that

Reply:
was able to do something along those lines to reduce test lines. thanks for the suggestion!